### DISCUSSION TITLE
Dynamically importing Javascript from URL or String and manipulating the module

### DISCUSSION THREAD

**Discussion by ten3roberts** [wasm-bindgen/wasm-bindgen#3539] [Q&A]
### Summary

Is it possible to import a ES module from a url or other downloaded text and `import` it through wasm.

Ideally, this will return an `Object` like `import` does, upon which you can dynamically reflect and call the exports or default export.

### Additional Details

The wasm file has no prior data on the javascript file, and can as such not be used with the `#[wasm_bindgen(module = )]` macro, as that requires the JS file to be known at compile time and will become globally available.

I would need a dynamic *local* object that is returned where and when it is needed

**Comment by daxpedda** [comment]
You could do it with `inline_js` of course, but not otherwise, see https://github.com/rustwasm/wasm-bindgen/issues/2865.
(this was broken by https://github.com/rustwasm/wasm-bindgen/pull/2855)
